### PR TITLE
XsuaaServiceConfigurationDefault supports access to other credentials

### LINKED ISSUE #802: XsuaaServiceConfigurationDefault breaking change since 2.11.1
Hi colleagues,

It seems that from version 2.11.1 of XSUAA Spring library, the `XsuaaServiceConfigurationDefault` class has breaking changes that affect the ability to get XSUAA VCAP properties that are not mapped in the class and do not have any getters. 

The `getProperty()` and `hasProperty()` methods were deleted.
For example, properties like API URL and tenant ID cannot be fetched now directly.

How can these properties be fetched via the XSUAA Spring library?

As far as I can see, the change is from here:
https://github.com/SAP/cloud-security-xsuaa-integration/pull/698/files#diff-578760e442614f47823e1a3c83ed22a0a8cab50f5005861b9417764499473606L115

Best regards,
Momchil

### PR DESCRIPTION
**Pull request by nenaraab**
relates to #627
closes #802

### CODE REVIEW

File: spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationFromEnvTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.sap.cloud.security.xsuaa;
+
+import org.junit.Before;
```
[the remark this answers is not in the document]
Comment (nenaraab) [reply spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationFromEnvTest.java:3]: I’ve chosen junit 4 as it works properly with environment test rule. Btw: this test code was deleted by mistake as part of 2.11.3 release. Furthermore, Spring-xsuaa has a lot of more junit 4 test cases, we can’t easily migrate.

[the remark this answers is not in the document]
Comment (liga-oz) [reply spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationFromEnvTest.java:3]: We agreed that all new test cases should be written in Junit5 as this appears new, that's why the comment and as provided in the link from previous comment, environment variables do work also in Junit5, but I leave it up to you to switch it over or not.

[the remark this answers is not in the document]
Comment (nenaraab) [reply spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationFromEnvTest.java:3]: Actually, as said - its no new code, its reactivation and adaption of code deleted with #698

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationDefault.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
 @Configuration
 public class XsuaaServiceConfigurationDefault implements XsuaaServiceConfiguration {
 
+	static final String VCAP_SERVICES_CREDENTIALS = "xsuaa credentials from VCAP_SERVICES environment";
```
Comment (liga-oz) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationDefault.java:18]: just my humble opinion on this message: when i see this description I need to link it with the NullPointer exception to understand that the xsuaa creds are missing, if i saw message that would have said "xsuaa creds ... are required" I wouldn't have a second thought on what's wrong, so I would suggest to add that "is required" or "must not be empty/null" at the end.

**Review by liga-oz — APPROVED** [review]
All in all looking good, therefore, approving, but not merging, just review my comment about the message and make changes or merge it. 😉

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationDefault.java (hunk 2 above)
Comment (nenaraab) [reply spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationDefault.java]: I haven't added it as it comes with "NullPointerException", anyhow....

**Pull request merged**